### ISSUE TITLE
Spherical crater example

### ISSUE THREAD

**Issue by sampotter** [sampotter/python-flux#49]

**Comment by sampotter** [comment]
Updated the spherical crater example, see PR #50 

I addressed Norbert's comments

I think I may have misinterpreted one of Norbert's points about comparing to the groundtruth matrix. It's true that we can do a few more direct comparisons. I added two new plots:

1. The relative l2 error for the temperature computed in the crater as computed by the groundtruth FF matrix and the compressed FF matrix for varying mesh fineness
2. The relative Frobenius error for the two matrices for varying mesh fineness

These should help us get a better sense of how well the compressed FF matrix approximates the groundtruth FF matrix. See below for `p = 5, .., 8`:

![FF_rel_fro_errors](https://user-images.githubusercontent.com/6035363/149563515-0f2bd0dd-8516-4d2f-8982-69f48de12f3b.png)
![h_vs_ptwise_errors](https://user-images.githubusercontent.com/6035363/149563557-786943d9-2ccc-45c2-85be-07c6a2e92d8f.png)

For the first plot, note that the choice of `eps` for compression was `1e-2`... Will be worthwhile checking what happens for other choices and see if the order of magnitude of the Frobenius norms follows suit.

For the second plot, note that the agreement between the two temperatures fields (and irradiance field) is very good. This error is much smaller than the large discretization error incurred by our approximation the integral equation using piecewise constant basis functions on each triangle.

**Comment by steo85it** [comment]
Very cool, thanks! I got a bit delayed with the tests by a stupid error on our cluster (showing an "exceeded quota" w/o any reason after a recent maintenance), but I'll compute them as soon as IT solves the issue.

**Comment by sampotter** [comment]
OK, the spherical crater example is basically done. The important data is being collected and plots are being made.

Running `examples/spherical_crater/run_example.sh` will generate a directory `examples/spherical_crater/spherical_crater_plots` which can be dropped into our Overleaf document (I also added some figure placeholders there, as well).

I am going to close this. Let me know when you're ready and we can re-run a big test on your cluster and make some real plots for the paper.

**Comment by steo85it** [comment]
I'm running them now (using the hemispherical_crater branch). Should I pull and re-run?

**Comment by sampotter** [comment]
Yes... but hold on, I thought I had pushed my latest changes but didn't. Now I am fixing some conflicts.

**Comment by sampotter** [comment]
OK, please try using my latest commit to master.

If you run into any trouble, let me know. I will keep this issue open until you successfully run it on your cluster.

Please see `PMIN` and `PMAX` in `examples/spherical_crater/run_example.sh`. I would first try running it with `PMAX` set to something relatively small, like 7 or 8. You should also set the `P` value in `collect_memory_usage_stats.sh` to a similarly small value. Then `run_example.sh` should complete in a few minutes and you can verify that everything runs OK and the plots are created. After that it will be a game to see find the largest value of `p` is that can be run on your cluster. :-)

**Comment by sampotter** [comment]
Incidentally, to find that value, I would just run `ingersoll.py` with different values passed to the `-p` flag, THEN run `run_example.sh` with that choice. :-)

**Comment by steo85it** [comment]
Super! I'm actually running `run_example.sh` in a slightly different way to send each `p` to a different machine on the cluster, but same concept. :) 
Your latest commit seems to work fine for me (some scripts are failing because of this when p=4, but that's expected). Here are the results of `collect_ingersoll_results.sh` for p=4,8 (p={10..20..2} are running, just to see how far can I get):
```
1.e-2
p = 4
- groundtruth temperature in shadow: 148.86 K
- created tri. mesh with 244 faces
- wrote ingersoll.obj
- finished assembly (0.1 Mb) [0.31 s]
- wrote FF.bin
- computed direct irradiance [0.00 s]
- wrote E.png to disk
- computed radiosity [0.00 s]
- wrote B.png
- computed T [0.00 s]
- wrote T.png
- wrote error.png
- max error: 4.74 K
- RMS error: 1.47 K
- wrote B.bin
- wrote T.bin
- wrote stats.json
real	0m4.741s
user	0m5.832s
sys	0m4.840s
```
and
```
1.e-2
p = 8
- groundtruth temperature in shadow: 148.86 K
- created tri. mesh with 6089 faces
- wrote ingersoll.obj
- finished assembly (4.4 Mb) [64.60 s]
- wrote FF.bin
- computed direct irradiance [0.02 s]
- wrote E.png to disk
- computed radiosity [0.05 s]
- wrote B.png
- computed T [0.14 s]
- wrote T.png
- wrote error.png
- max error: 1.27 K
- RMS error: 0.33 K
- wrote B.bin
- wrote T.bin
- wrote stats.json

real	1m12.049s
user	3m55.272s
sys	0m46.159s
```
